### PR TITLE
new python package spectra: init at 0.0.11

### PR DESCRIPTION
**Pull request by dflores1**
New package for bioinformatics Multiqc at 1.25.1
New python package spectra at 0.0.11 (multiqc dependency)

https://github.com/jsvine/spectra/tree/master

## Things done

- Built on platform(s)
  - [*] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [*] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [*] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [*] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/spectra/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  pytestCheckHook,
+  nose,
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/spectra/default.nix:7]: nose was removed in #348699

File: pkgs/development/python-modules/spectra/default.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    pytestCheckHook
+    nose
+  ];
+
+  pytestFlagsArray = [ "-W 'ignore:pkg_resources is deprecated as an API:DeprecationWarning'" ];
+
+  pythonImportsCheck = [ "spectra" ];
+
+  meta = {
+    description = "Spectra is a Python library that makes color math, color scales, and color-space conversion easy";
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/spectra/default.nix:39]:
```suggestion
    description = "Python library for color math, color scales, and color-space conversion";
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/mu/multiqc/package.nix [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+  meta = {
+    description = "Aggregate bioinformatics results across many samples into a single report";
+    longDescription = ''
+      MultiQC is a tool to create a single report with interactive plots for multiple bioinformatics analyses across many samples.
+
+      Reports are generated by scanning given directories for recognised log files. These are parsed and a single HTML report is generated summarising the statistics for all logs found. MultiQC reports can describe multiple analysis steps and large numbers of samples within a single plot, and multiple analysis tools making it ideal for routine fast quality control.
+    '';
+    homepage = "https://multiqc.info";
+    changelog = "https://github.com/MultiQC/MultiQC/blob/main/CHANGELOG.md";
```
Comment (pluiedev) [inline pkgs/by-name/mu/multiqc/package.nix:114]: Changelog should point to the current, not latest version
```suggestion
    changelog = "https://github.com/MultiQC/MultiQC/blob/refs/tags/${version}/CHANGELOG.md";
```

File: pkgs/by-name/mu/multiqc/package.nix [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    sourceProvenance = [ lib.sourceTypes.fromSource ];
```
Comment (pluiedev) [inline pkgs/by-name/mu/multiqc/package.nix:115]: This is the default anyway
```suggestion
```

File: pkgs/by-name/mu/multiqc/package.nix [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  disabled = python3Packages.pythonOlder "3.8";
+
+  # Two data sources. One for the code, another for the test data
+  # thanks @rkoe. https://discourse.nixos.org/t/how-to-create-package-with-multiple-sources/9308/2
+  srcs = [
+    (fetchFromGitHub {
+      name = pname;
+      owner = "MultiQC";
+      repo = "MultiQC";
+      rev = "v${version}";
```
Comment (pluiedev) [inline pkgs/by-name/mu/multiqc/package.nix:24]:
```suggestion
      rev = "refs/tags/v${version}";
```

File: pkgs/development/python-modules/spectra/default.nix [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      * Color ranges
+      * Color blending
+      * Brightening/darkening colors
+      * Saturating/desaturating colors
+      * Conversion to/from multiple color spaces
+
+      Spectra is built on colormath and grapefruit. Spectra is enormously inspired by chroma.js and d3's scales.
+    '';
+    homepage = "https://github.com/jsvine/spectra";
+    sourceProvenance = [ lib.sourceTypes.fromSource ];
```
Comment (pluiedev) [inline pkgs/development/python-modules/spectra/default.nix:53]:
```suggestion
```

File: pkgs/by-name/bu/bustools/package.nix [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+  buildInputs = [ zlib ];
+
+  passthru.tests.version = testers.testVersion {
+    package = bustools;
+    command = "bustools version";
+  };
+
+  meta = {
+    description = "bustools is a program for manipulating BUS files for single cell RNA-Seq datasets";
```
Comment (pluiedev) [inline pkgs/by-name/bu/bustools/package.nix:32]:
```suggestion
    description = "Program for manipulating BUS files for single cell RNA-Seq datasets";
```

File: pkgs/by-name/bu/bustools/package.nix [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "bustools";
+  version = "0.44.0";
+
+  src = fetchFromGitHub {
+    owner = "BUStools";
+    repo = "bustools";
+    rev = "v${version}";
```
Comment (pluiedev) [inline pkgs/by-name/bu/bustools/package.nix:18]:
```suggestion
    rev = "refs/tags/v${version}";
```

**Review by pluiedev — COMMENTED** [review]
PR title should be changed to something like `bustools: init at 0.44.0, multiqc: init at 1.25.1`. Spectra probably doesn't need to be mentioned as it is a dependency of `multiqc`

**Pull request closed without merging**